django/db/backends/base/schema.py

```diff
@@ -396,6 +402,28 @@ def _column_default_sql(self, field):
         """
         return "%s"
 
+    def db_default_sql(self, field):
+        """Return the sql and params for the field's database default."""
+        query = Query(model=field.model)
+        compiler = query.get_compiler(connection=self.connection)
+        default_sql, params = compiler.compile(field.db_default)
+        wrapper_sql = self._wrapper_sql_for_default(field.db_default)
+        return self._format_sql_for_default(wrapper_sql, default_sql, params)
+
+    def _wrapper_sql_for_default(self, default):
+        """
+        Return the wrapper sql required for database defaults.
+
+        Database functions require a slightly different syntax to literals.
+        """
+        from django.db.models.expressions import Value
+
+        return "%s" if isinstance(default, Value) else "(%s)"
```

Comment:
aha, maybe this could also work for `Collate`? should there be an expression-level hook instead?

Reply:
This did not work for `Collate`. I'm not sure it's worth an expression-level hook if it's only relevant to `Value`.